src/module/functions/local_function/mod.rs

```diff
@@ -794,102 +831,298 @@ fn validate_instruction(
             ctx.unreachable(expr);
         }
 
-        Operator::MemorySize { reserved } => {
+        O::MemorySize { reserved } => {
             if reserved != 0 {
                 bail!("reserved byte isn't zero");
             }
             let memory = ctx.indices.get_memory(0)?;
             let expr = ctx.func.alloc(MemorySize { memory });
-            ctx.push_operand(Some(ValType::I32), expr);
+            ctx.push_operand(Some(I32), expr);
         }
-        Operator::MemoryGrow { reserved } => {
+        O::MemoryGrow { reserved } => {
             if reserved != 0 {
                 bail!("reserved byte isn't zero");
             }
-            let (_, pages) = ctx.pop_operand_expected(Some(ValType::I32))?;
+            let (_, pages) = ctx.pop_operand_expected(Some(I32))?;
             let memory = ctx.indices.get_memory(0)?;
             let expr = ctx.func.alloc(MemoryGrow { memory, pages });
-            ctx.push_operand(Some(ValType::I32), expr);
+            ctx.push_operand(Some(I32), expr);
         }
 
-        Operator::Nop => {}
-
-        Operator::I32Load { memarg } => load(ctx, memarg, ValType::I32, LoadKind::I32)?,
-        Operator::I64Load { memarg } => load(ctx, memarg, ValType::I64, LoadKind::I64)?,
-        Operator::F32Load { memarg } => load(ctx, memarg, ValType::F32, LoadKind::F32)?,
-        Operator::F64Load { memarg } => load(ctx, memarg, ValType::F64, LoadKind::F64)?,
-        Operator::V128Load { memarg } => load(ctx, memarg, ValType::V128, LoadKind::V128)?,
-        Operator::I32Load8S { memarg } => load(
-            ctx,
-            memarg,
-            ValType::I32,
-            LoadKind::I32_8 { sign_extend: true },
-        )?,
-        Operator::I32Load8U { memarg } => load(
-            ctx,
-            memarg,
-            ValType::I32,
-            LoadKind::I32_8 { sign_extend: false },
-        )?,
-        Operator::I32Load16S { memarg } => load(
-            ctx,
-            memarg,
-            ValType::I32,
-            LoadKind::I32_16 { sign_extend: true },
-        )?,
-        Operator::I32Load16U { memarg } => load(
-            ctx,
-            memarg,
-            ValType::I32,
-            LoadKind::I32_16 { sign_extend: false },
-        )?,
-        Operator::I64Load8S { memarg } => load(
+        O::Nop => {}
```

Comment:
Not convinced `Operator` -> `O` is really a win, but whatever...

Reply:
I largely wanted to makeit so `cargo fmt` preserved reasonably readable code here and didn't spew all the parsing over hundreds of lines, but I don't think I was overly successful. I'll switch it back